doc/neps/ufunc-overrides.rst

```diff
@@ -106,163 +116,193 @@ Returning ``NotImplemented`` to user should not happen. Moreover::
 Here, it appears that the sparse matrix was converted to an object array
 scalar, which was then multiplied with all elements of the ``b`` array.
 However, this behavior is more confusing than useful, and having a
-``TypeError`` would be preferable.
+:exc:`TypeError` would be preferable.
 
-Adding the ``__numpy_ufunc__`` functionality fixes this and would
+Adding the ``__array_ufunc__`` functionality fixes this and would
 deprecate the other ufunc modifying functions.
 
 .. [5] http://mail.python.org/pipermail/numpy-discussion/2011-June/056945.html
 
+.. [6] https://github.com/numpy/numpy/issues/5844
 
 Proposed interface
 ==================
 
-Objects that want to override Ufuncs can define a ``__numpy_ufunc__`` method.
-The method signature is::
+The standard array class :class:`ndarray` gains an ``__array_ufunc__``
+method and objects can override Ufuncs by overriding this method (if
+they are :class:`ndarray` subclasses) or defining their own. The method
+signature is::
 
-    def __numpy_ufunc__(self, ufunc, method, i, inputs, **kwargs)
+    def __array_ufunc__(self, ufunc, method, *inputs, **kwargs)
 
 Here:
 
 - *ufunc* is the ufunc object that was called. 
 - *method* is a string indicating which Ufunc method was called
   (one of ``"__call__"``, ``"reduce"``, ``"reduceat"``,
   ``"accumulate"``, ``"outer"``, ``"inner"``). 
```

Comment:
Sorry, did not read right: there is no `inner` that I know of, i.e., another good catch!